hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3285,6 +3323,66 @@ func reconcileMachineApproverDeployment(deployment *appsv1.Deployment, hc *hyper
 	return nil
 }
 
+func reconcileOpenshiftIngressNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyperv1.HostedCluster) error {
+	policy.Spec.Ingress = []networkingv1.NetworkPolicyIngressRule{
+		{
+			From: []networkingv1.NetworkPolicyPeer{
+				{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"network.openshift.io/policy-group": "ingress",
+						},
+					},
+				},
+			},
+		},
+	}
+	policy.Spec.PodSelector = metav1.LabelSelector{}
+	policy.Spec.PolicyTypes = []networkingv1.PolicyType{networkingv1.PolicyTypeIngress}
+	policy.Annotations[hostedClusterAnnotation] = client.ObjectKeyFromObject(hcluster).String()
```

Comment:
Nit: This isn't needed because it gets inserted in CreateOrUpdate for everything: https://github.com/openshift/hypershift/blob/55b0480574d150eeb3c15f2a0c4954caac62ecd1/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L165